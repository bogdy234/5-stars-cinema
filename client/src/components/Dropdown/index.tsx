import { FC, ReactElement, useRef } from "react";
import useClickOutside from "../../utils/hooks/useClickOutside";

interface DropdownProps {
  show: boolean;
  options: string[];
  onClickOption: (option: string) => void;
  hide: () => void;
}

const Dropdown: FC<DropdownProps> = ({
  show,
  options,
  onClickOption,
  hide,
}): ReactElement => {
  const ref = useRef(null);

  useClickOutside(ref, hide);
  return (
    <div
      id="dropdown"
      className={`${
        show ? "" : "hidden"
      } z-10 w-44 bg-white rounded divide-y divide-gray-100 shadow dark:bg-gray-700 absolute top-0`}
      ref={ref}
    >
      <ul
        className="py-1 text-sm text-gray-700 dark:text-gray-200"
        aria-labelledby="dropdownDefault"
      >
        {options &&
          options.map((option) => (
            <li key={option}>
              <div
                className="block py-2 px-4 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
                onClick={() => onClickOption(option)}
              >
                {option}
              </div>
            </li>
          ))}
      </ul>
    </div>
  );
};

export default Dropdown;
