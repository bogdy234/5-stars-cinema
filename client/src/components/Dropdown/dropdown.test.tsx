import renderer from "react-test-renderer";
import Dropdown from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Dropdown
        show
        options={["1", "2"]}
        onClickOption={jest.fn()}
        hide={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
