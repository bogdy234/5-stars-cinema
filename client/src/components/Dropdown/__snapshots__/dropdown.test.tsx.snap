// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" z-10 w-44 bg-white rounded divide-y divide-gray-100 shadow dark:bg-gray-700 absolute top-0"
  id="dropdown"
>
  <ul
    aria-labelledby="dropdownDefault"
    className="py-1 text-sm text-gray-700 dark:text-gray-200"
  >
    <li>
      <div
        className="block py-2 px-4 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
        onClick={[Function]}
      >
        1
      </div>
    </li>
    <li>
      <div
        className="block py-2 px-4 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
        onClick={[Function]}
      >
        2
      </div>
    </li>
  </ul>
</div>
`;
